scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/GraphQLQueries.java

```diff
@@ -135,29 +122,81 @@ public Query getRevisionAndjobsInfoQuery(final String user, final boolean pendin
             if (fetch) {
                 if (filterModel.isMatchAny() && !filterModel.getConstraints().isEmpty()) {
                     for (Constraint constraint : filterModel.getConstraints()) {
-                        input.add(getJobInputWithStatus(status,
-                                                        user,
-                                                        constraint.getFilteringString(Field.ID),
-                                                        constraint.getFilteringString(Field.PRIORITY),
-                                                        constraint.getFilteringString(Field.NAME)));
+                        input.add(getJobInput(status, user, Collections.singletonList(constraint)));
                     }
                 } else {
                     String id = null;
                     String priority = null;
                     String name = null;
 
-                    for (Constraint contraint : filterModel.getConstraints()) {
-                        id = getValue(id, contraint.getFilteringString(Field.ID));
-                        priority = getValue(priority, contraint.getFilteringString(Field.PRIORITY));
-                        name = getValue(name, contraint.getFilteringString(Field.NAME));
-                    }
-                    input.add(getJobInputWithStatus(status, user, id, priority, name));
+                    input.add(getJobInput(status, user, filterModel.getConstraints()));
                 }
             }
         }
         return input;
     }
 
+    private JobInput getJobInput(JobStatus status, String user, List<Constraint> constraints) {
+        JobInput.Builder input = new JobInput.Builder();
+        input.status(status.name().toUpperCase());
+        if (user != null)
+            input.owner(user);
+
+        String id = null;
+        String priority = null;
+        String name = null;
+        for (Constraint constraint : constraints) {
+            String value = constraint.getValue();
+            switch (constraint.getTargetField()) {
+                case ID: {
+                    switch (constraint.getAction()) {
+                        case EQUALS:
+                            id = getValue(id, value);
+                            break;
+                        case GREATER_THAN_OR_EQUAL_TO:
+                            input.greaterThanId(value);
+                            break;
+                        case LESS_THAN_OR_EQUAL_TO:
+                            input.lowerThanId(value);
+                            break;
+                        default:
+                            ;
```

Comment:
Can you add a break for the default case in order to keep consistency in the code